browser/src/control/jsdialog/Util.KeyboardGridNavigation.ts

```diff
@@ -0,0 +1,153 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * JSDialog.ScrollableBar - helper for creating toolbars with scrolling left/right
+ */
+
+declare var JSDialog: any;
+
+function handleKeyboardNavigation(
+	event: KeyboardEvent,
+	currentElement: HTMLElement,
+	parentContainer: HTMLElement,
+	handleSelection?: any,
+	builder?: any,
+	widgetData?: any,
+	palette?: any,
+) {
+	switch (event.key) {
+		case 'Enter':
+		case ' ':
+			if (handleSelection && !(currentElement instanceof HTMLSelectElement)) {
+				handleSelection(currentElement, builder, widgetData, palette);
```

Comment:
palette is specific for the color picker, can we somehow avoid it? so it will not confuse others if we will reuse this util?